common/src/main/java/org/corfudb/common/util/CompatibilityVectorUtils.java

```diff
@@ -0,0 +1,149 @@
+        return isFeatureEnabled(feature.getNumber(), vectors);
+    }
+
+    /**
+     * Checks if the feature is enabled in the given {@link ByteString} or not.
+     *
+     * @param feature int value of the feature to be checked
+     * @param vectors the {@link ByteString} that contains the bits of the Corfu features
+     * @return a boolean value that denotes if the feature is enabled or not.
+     */
+    public static boolean isFeatureEnabled(int feature, ByteString vectors) {
+        // if the feature number (0-indexed) is greater than or equal the bytes length,
+        // or if it is a negative value, then the feature is not enabled.
+        if (feature >= 8 * vectors.size() || feature < 0)
+            return false;
+
+        // check if the bit is set in the vector
+        // = ((corresponding byte) AND (1 only on the index of the bit) !=0)
+        return ((vectors.byteAt((feature) / 8) & (1 << ((feature) % 8))) != 0);
+    }
+
+    /**
+     * This enum loads when {@link #getCompatibilityVectors()} is called
+     * for the first time. It gives a thread-safe lazy-initialization because the class loader
+     * guarantees that all static initialization is complete before getting access to the class.
+     */
+    public enum Feature {
+        // Comma separated enum values indicating each feature along with their sequence number.
+        // FEATURE_A(0),
+        // FEATURE_B(1);
+        ;
+
+        /**
+         * Contains the vectors representing the Corfu features enabled so far.
+         */
+        private static final ByteString vectors = generateVectors();
+
+        /**
+         * Local map to store the integer sequencer numbers to Features map.
+         * This map will be used to get the feature enum from its sequence number in
+         * {@link #getFeatureOf(int)} method.
+         */
+        private static final Map<Integer, Feature> features_map = generateFeaturesMap();
+
+        private final int number;
+
+        /**
+         * A constructor to initiate the enum value along with its sequence number
+         *
+         * @param number the sequence number of the enum
+         */
+        Feature(int number) {
+            if (number < 0)
+                throw new IllegalArgumentException("Enum number should be a non negative integer." +
+                        " The current value is " + number + ".");
+            this.number = number;
+        }
+
+        private static ByteString generateVectors() {
+            Feature[] features = Feature.values();
+
+            // Though initializing with the length is optional, it improves performance.
+            BitSet bitSet = new BitSet(features.length);
+
+            for (Feature feature : features) {
+                bitSet.set(feature.getNumber());
+            }
+
+            log.info("generateVectors: Compatibility vectors are successfully initiated with {}",
+                    bitSet);
```

Comment:
Since this log will only be done once for a node, would it be better to log out the features also? such that we could have a comparison

Reply:
Will do, thanks!